jni/src/org_opensearch_knn_jni_FaissService.cpp

```diff
@@ -75,6 +75,24 @@ JNIEXPORT void JNICALL Java_org_opensearch_knn_jni_FaissService_createBinaryInde
     }
 }
 
+JNIEXPORT void JNICALL Java_org_opensearch_knn_jni_FaissService_createByteIndex(JNIEnv * env, jclass cls, jintArray idsJ,
+                                                                            jlong vectorsAddressJ, jint dimJ,
+                                                                            jstring indexPathJ, jobject parametersJ)
+{
+    try {
+        std::unique_ptr<knn_jni::faiss_wrapper::FaissMethods> faissMethods(new knn_jni::faiss_wrapper::FaissMethods());
```

Comment:
let's use std::make_unique to create objects which is already given for unique pointers

```
 auto faissMethods = std::make_unique<knn_jni::faiss_wrapper::FaissMethods>();
```

Reply:
discussed offline... `std::make_unique` is supported only from C++14